### ISSUE TITLE
Use native desktop notifications when somebody calls me

### ISSUE THREAD

**Issue by levinelson** [speak/issues#168]
I use multiple spaces / full screen apps on my Mac, so when somebody calls and I hear the ringtone, I don't necessarily know where to click to answer. The window might be on a desktop I'm not currently viewing, so I usually have to swipe to find it an answer.

**Comment by tommoor** [comment]
Hey Levi, that certainly is strange - the incoming call box is designed to appear on every workspace in the same fashion as the dock. Does the dock only appear on one screen for you also?

**Comment by levinelson** [comment]
Ah, gotcha. Yep, the dock and call box doesn't appear over any apps running in fullscreen mode, at least on Yosemite. It does appear on every normal workspace, like you said.

So to clarify, the scenario where I felt disoriented was when I was running my text editor in full screen. I heard a ringing sound and didn't know what it was, so I swiped around until I saw the speak dialog box on one of my normal desktop workspaces and realized what was happening.

**Comment by tommoor** [comment]
It does seem as though this is a limitation of OSX, we could copy Skype here and jump the icon in the dock - would this be an improvement?

**Comment by levinelson** [comment]
Hey Tom, I just tested Slack's (native OSX) notifications while running an app in full screen, and I was able to see the notification come over the app just fine (I'm also running El Capitan now if that makes a difference). I wonder if it has something to do with that being the desktop I was actively focused in? I'm not sure bouncing the dock would be much help here because I don't see the dock on every screen or fullscreen app either.

**Comment by tommoor** [comment]
Yes the only thing that can appear on top is a native notification, unfortunately these can't be customized in the way we need to to include call controls, avatars and the other details :smile:

This one might be a "won't/can't fix" I'm afraid

**Comment by levinelson** [comment]
Understandable!

**Comment by levinelson** [comment]
Update: just noticed that I *can* see a bouncing dock icon when in a full screen app. So, that actually might be a good idea to at least signal that there's something going on with Speak!
